### PR TITLE
HybridCache: relocate to dotnet/extensions

### PR DESCRIPTION
**Pull request by mgravell**
# `HybridCache`: relocate to dotnet/extensions

Plan agreed in "tactics", 2024-08-27; `HybridCache` is jumping from dotnet/aspnetcore to dotnet/extensions and will not ship in the framework build of NET 9 RC1+. The library-specific NET 9 RC1 ship-stop is manual; this PR is to clean up for NET 9 RC2 onwards.

- remove src/Caching/Hybrid (leaves relocation README.md)
- remove from sln[f]
- remove from eng files (via automated rebuild)
- remove minimal usage from tangential benchmark project

(this is https://github.com/dotnet/aspnetcore/pull/57611 but rebased to 9.0, after fighting git for a bit)

### CODE REVIEW

**Comment by mgravell** [comment]
Build code check is because of shipped/unshipped delta

File: src/Caching/Hybrid/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# `HybridCache`
```
Comment (BrennanConroy) [inline src/Caching/Hybrid/README.md:1]: It's a nice thought, I personally wouldn't leave this in though

**Review by BrennanConroy — APPROVED** [review]

File: src/Caching/Hybrid/README.md (hunk 1 above)
Comment (mgravell) [reply src/Caching/Hybrid/README.md]: fine, nuked in 76d9143b81174b850b5212ab33f2a9af4ba9082e

**Comment by wtgodbe** [comment]
Code Check failures are expected as this is in a release branch

**Comment by mgravell** [comment]
@wtgodbe  are you able to merge this removal? `HybridCache` has now been relocated to [dotnet/extensions:dev](https://github.com/dotnet/extensions/tree/dev/src/Libraries/Microsoft.Extensions.Caching.Hybrid)

**Pull request merged**